### PR TITLE
Add tsfresh 0.18.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Popularity:  49641 downloads -  tsfresh  0.18.0  Automatic extraction of relevant features from time series
  license: MIT
Release date:  Mar 6, 2021
Bug Tracker: new open issues https://github.com/blue-yonder/tsfresh/issues
Github releases:  https://github.com/blue-yonder/tsfresh/releases
Upstream Changelog: https://github.com/blue-yonder/tsfresh/blob/main/CHANGES.rst
Upstream license file: https://github.com/blue-yonder/tsfresh/blob/v0.18.0/LICENSE.txt
Requirements from conda-forge: https://github.com/conda-forge/tsfresh-feedstock/blob/master/recipe/meta.yaml
Upstream requirements: https://github.com/blue-yonder/tsfresh/blob/v0.18.0/requirements.txt
Upstream setup file: https://github.com/blue-yonder/tsfresh/blob/v0.18.0/setup.py
Upstream setup.cfg: https://github.com/blue-yonder/tsfresh/blob/v0.18.0/setup.cfg

Actions:
1. Add entry_points
2. Update dependencies
3. Disable pip check

Result:
- all-succeeded on concourse

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
We are still waiting when `matrixprofile`'s artifacts will be uploaded on  defaults for`linux-aarch64`, `linux-s390x`, `osx-arm64` https://github.com/AnacondaRecipes/matrixprofile-feedstock/pull/1

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,34 +15,39 @@ build:
   number: 0
   noarch: python
   script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
+  entry_points:
+    - run_tsfresh = tsfresh.scripts.run_tsfresh:main
```
Comment (katietz) [inline recipe/meta.yaml:21]: We need to add a Jira ticket for osx-arm64 (and linux-aarch64?) and skip those targets (plus a comment)

**Review by katietz — CHANGES_REQUESTED** [review]

**Comment by dbast** [comment]
s390x error: dependency missing `- matrixprofile[version='>=1.1.10,<2.0.0']`

osx-arm64 error: dependency missing `- statsmodels[version='>=0.9.0'] and  - matrixprofile[version='>=1.1.10,<2.0.0']`

linux-aarch64: agent broken... I'll look into that.

**Review by katietz — APPROVED** [review]
Please don't forget to file a Jira ticket for it. So we don't miss it

**Comment by skupr-anaconda** [comment]
JIRA ticket created: tsfresh 0.18.0 on linux-arm64

**Comment by skupr-anaconda** [comment]
`tsfresh` failed on `s390x` because we can’t build its dependency `stumpy 1.9.2` on this platform, see https://github.com/AnacondaRecipes/stumpy-feedstock/pull/1#issuecomment-935841894

**Pull request merged**